### ISSUE TITLE
Code compilation error

### ISSUE THREAD

**Issue by iainsymthe** [vincentcasseau/hyStrath#65]
I was trying a complete reinstall of hyStrath using 
git clone https://github.com/vincentcasseau/hyStrath.git

I get compilation errors that look as shown below

%install.sh
-----  INSTALLATION  -----
1 - CFD module
2 - DSMC module
3 - Hybrid PIC-DSMC module
4 - CFD-MHD module
5 - All modules

-----  SYNCHRONISATION  -----
11 - CFD module
12 - DSMC module
13 - Hybrid PIC-DSMC module
14 - CFD-MHD module
15 - All modules

Enter choice: 5
[>>>>>>>>>>>>>>>>>>>>] 100% installed CFD module
SUCCESS
./install.sh: line 226: wait: -n: invalid option
wait: usage: wait [id]
[>>>>>>>>>>>>>>>>>>>>] 100% installed DSMC module
FAIL: check logInstall-DSMC
./install.sh: line 228: wait: -n: invalid option
wait: usage: wait [id]
[>>>>>>>>>>>>>>>>>>>>] 100% installed hybrid PIC-DSMC module
FAIL: check logInstall-hybridPICDSMC
./install.sh: line 230: wait: -n: invalid option
wait: usage: wait [id]
[>>>>>>>>>>>>>>>>>>>>] 100% installed CFD-MHD module
SUCCESS

The end of log-DSMCFoam  looks like

from macroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcVolFieldsFromFields.C:35:
macroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcVolFieldsFromFields.C:280:20: error: ‘
rhoNMean_’ was not declared in this scope
             forAll(rhoNMean_, celli)
                    ^
reinstall_OpenFoam_hyStrath/OpenFOAM-v1706/src/OpenFOAM/lnInclude/stdFoam.H:154:30: note: in definition of macro ‘forAll’
     for (Foam::label i=0; i<(list).size(); ++i)
                              ^~~~
macroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcVolFieldsFromFields.C:280:20: note: su
ggested alternative: ‘dsmcNMean_’
             forAll(rhoNMean_, celli)
                    ^
/reinstall_OpenFoam_hyStrath/OpenFOAM-v1706/src/OpenFOAM/lnInclude/s
tdFoam.H:154:30: note: in definition of macro ‘forAll’
     for (Foam::label i=0; i<(list).size(); ++i)
                              ^~~~
make: *** [Make/linux64GccDPInt32Opt/macroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcV
olFieldsFromFields.o] Error 1

**Comment by vincentcasseau** [comment]
Hi Iain,

That's fixed, thanks for reporting it.

Side note: what's your version of bash?
`bash --version`

`wait -n` was introduced in bash 4.3 (~ 2014). I am surprised you don't have it.

**Comment by iainsymthe** [comment]
Thanks, Vincent.
I do have an older version of bash
bash --versionGNU bash, version 4.2.46(2)-release (x86_64-redhat-linux-gnu)
To get the DSMC modules recompiled do I just do install.sh and pick only the DSMC option
I tried to get the DSMC module recompiled again in my reinstall directory by doing 

%install.sh 4-----  INSTALLATION  -----1 - CFD module2 - DSMC module3 - Hybrid PIC-DSMC module4 - CFD-MHD module5 - All modules
-----  SYNCHRONISATION  -----11 - CFD module12 - DSMC module13 - Hybrid PIC-DSMC module14 - CFD-MHD module15 - All modules
Enter choice: 2[>>>>>>>>>           ]  49% installing DSMC module./install.sh: line 78:  1785 Terminated              progress_bar $sleep_period "installing DSMC module"[>>>>>>>>>>>>>>>>>>>>] 100% installed DSMC moduleFAIL: check logInstall-DSMC

The errors in logInstall-DSMC are as follows:
macroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcVolFieldsFromFields.C:145:44: error: ‘vibrationalETotal_’ was not declared in this scopemacroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcVolFieldsFromFields.C:157:19: error: ‘rotationalEMean_’ was not declared in this scopemacroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcVolFieldsFromFields.C:209:16: error: ‘rhoNMean_’ was not declared in this scopemacroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcVolFieldsFromFields.C:213:69: error: ‘mccu_’ was not declared in this scopemacroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcVolFieldsFromFields.C:214:23: error: ‘mcc_’ was not declared in this scopemacroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcVolFieldsFromFields.C:220:23: error: ‘rotationalEMean_’ was not declared in this scopemacroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcVolFieldsFromFields.C:231:69: error: ‘mccv_’ was not declared in this scopemacroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcVolFieldsFromFields.C:248:69: error: ‘mccw_’ was not declared in this scopemacroscopicProperties/derived/combined/dsmcVolFieldsFromFields/dsmcVolFieldsFromFields.C:280:20: error: ‘rhoNMean_’ was not declared in this scope


    On Tuesday, December 29, 2020, 04:35:21 PM CST, Vince <notifications@github.com> wrote:  
 
 


Hi Iain,

That's fixed, thanks for reporting it.

Side note: what's your version of bash?
bash --version

wait -n was introduced in bash 4.3 (~ 2014). I am surprised you don't have it.

**Comment by vincentcasseau** [comment]
Hi Iain,

I am sure I fixed it in commit 079c55e.
Could you please type:
```
git pull origin master
git log
```

and check that you see that at the top

```
commit 914694ffc94f14cbea5074cf1364e38218e7adf2 (HEAD -> master, origin/master, origin/HEAD)
Author: vincentcasseau <hystrath@gmail.com>
Date:   Tue Dec 29 19:52:01 2020 -0500

    hy2Foam: removing untested numerical schemes (follow-up of 7a61f7d)
    and adding mesh motion to the reformated KNP and KT files


```

and finally 
`./install.sh 4 2>/dev/null`

Thanks

**Comment by iainsymthe** [comment]
Thanks, Vincent.  I verified the header in git log to make sure everything was OK and now I can compile both the DSMC and PIC DSMC modules.  I also did a complete resync (option 15) as well.

Has any regression testing been done to see if the new changes have not altered results run with the older version of the code?
    On Wednesday, December 30, 2020, 08:40:46 AM CST, Vince <notifications@github.com> wrote:  
 
 


Hi Iain,

I am sure I fixed it in commit 079c55e.
Could you please type:
git pull origin master
git log

and check that you see that at the top
commit 914694ffc94f14cbea5074cf1364e38218e7adf2 (HEAD -> master, origin/master, origin/HEAD)
Author: vincentcasseau <hystrath@gmail.com>
Date:   Tue Dec 29 19:52:01 2020 -0500

    hy2Foam: removing untested numerical schemes (follow-up of 7a61f7d)
    and adding mesh motion to the reformated KNP and KT files



and finally
./install.sh 4 2>/dev/null

Thanks
